src/components/NewPost/NewPost.js

```diff
@@ -20,22 +20,22 @@ const NewPost = ({setNewPostToggle}) => {
         })
         .then(res=>res.json())
         .then(data => {
-            console.log(data.secure_url)
             const imgURL = data.secure_url;
-            console.log(imgURL, description, user._id, 'req body')
             fetch('/api/post', {
                 method: 'POST',
                 headers: {'content-type': 'application/json'},
                 body: JSON.stringify({
                     imgURL: imgURL,
                     description: description,
-                    _id: user._id,
+                    user_id: user._id,
+                    username: user.username
                 })
             })
             .then(res => res.json())
             .then(data => {
                 if(data.status === 200) {
                     setNewPostToggle(false)
+                    setFeedRefresher(feedRefresher + 1)
                 } else {
                     console.log('SOMETHING WENT WRONG PLEASE TRY AGAIN')
                 }
```

Comment:
For this kind of code (and also for your backend handlers) where there are nested promise I would try to use `async/await` syntax as the readibility is somehow better having only one lvl of indentation